### ISSUE TITLE
reverting on incorrect .java files

### ISSUE THREAD

**Issue by Charmik** [INRIA/spoon#2946]
Hello! I am trying to parse .java files, but I have some files which were decompiled with synthetic methods. example:

```
@Retention(RetentionPolicy.RUNTIME)
@Target({ ElementType.ANNOTATION_TYPE })
public @interface ListenerClass {
    ListenerMethod[] method() default {};
    String remover() default "";

    public enum NONE {
        private static final /* synthetic */ NONE[] $VALUES;
        static {
            $VALUES = new NONE[0];
        }
        public static NONE valueOf(final String s) {
            return Enum.valueOf(NONE.class, s);
        }
        public static NONE[] values() {
            return NONE.$VALUES.clone();
        }
    }
}

```
it has synthetic enum methods, so it doesn't compile.
I try to parse it by: 
Launcher launcher = new Launcher();
launcher.addInputResource(source.toAbsolutePath().toString());
CtModel model = launcher.buildModel(); 

it gives me:
java.lang.IllegalArgumentException: The method remover() should be a spoon.reflect.declaration.CtAnnotationMethod

it breaks in CtAnnotationTypeImpl.addMethod because method has a type "CtMethodImpl", not a "CtAnnotationMethod".

but if you swap 2 lines: 
String remover() default "";
ListenerMethod[] method() default {};

it works. Can you suggest what I can do with not really good .java files? Can I just skip places which doesn't compile? I didn't find any options for it. I think it works on jdt-eclipse. Thanks!

**Comment by surli** [comment]
Hi @Charmik 

I've the feeling that there is two different issues here:
  - the synthetic enum
  - the two methods that are not in the right order

Could you confirm that? By checking if it fails if you remove the synthetic enum? 

> Can you suggest what I can do with not really good .java files? Can I just skip places which doesn't compile?

I don't think we can skip compilation error, now you can build the model without the erroring files: by default, Spoon is in "no classpath mode", which means it will work even if not all sources files are given as input.

**Comment by Charmik** [comment]
> Hi @Charmik
> 
> I've the feeling that there is two different issues here:
> 
> * the synthetic enum
> * the two methods that are not in the right order
> 
> Could you confirm that? By checking if it fails if you remove the synthetic enum?
> 
> > Can you suggest what I can do with not really good .java files? Can I just skip places which doesn't compile?
> 
> I don't think we can skip compilation error, now you can build the model without the erroring files: by default, Spoon is in "no classpath mode", which means it will work even if not all sources files are given as input.

yes, it works perfectly without enum.
It is not about class path. I have all classes (ListenerMethod is just another annotation). It's about synthetic methods in enum, this class doesn't compile at all. I would like to get
ListenerMethod[] method() default {};
and
String remover() default "";
but without enum, because mistakes inside enum. Can spoon help me with it? or he just says IllegalArgument and that's all ?

**Comment by surli** [comment]
> It is not about class path. 

I understood that, it was simply a suggest of a workaround there: to not include the failing java class as input for Spoon :) 

> Can spoon help me with it? or he just says IllegalArgument and that's all ?

I guess there's a bug in Spoon for this specific case. We'll need to reproduce and fix. If you want you can propose a PR with a failing test for Spoon, it'll help us a lot for fixing it.

**Comment by Charmik** [comment]
> > It is not about class path.
> 
> I understood that, it was simply a suggest of a workaround there: to not include the failing java class as input for Spoon :)
> 
> > Can spoon help me with it? or he just says IllegalArgument and that's all ?
> 
> I guess there's a bug in Spoon for this specific case. We'll need to reproduce and fix. If you want you can propose a PR with a failing test for Spoon, it'll help us a lot for fixing it.

I created PR:
https://github.com/INRIA/spoon/pull/2947

**Comment by nharrand** [comment]
Is this resolved by #2947 ?

**Comment by Egor18** [comment]
This issue is already resolved by #2947. I think it can be closed now.
